contracts/Holdable.sol

```diff
@@ -0,0 +1,240 @@
+        require(block.timestamp < holds[operationId].expiration || holds[operationId].expires == false, "This hold has already expired");
+        require(holds[operationId].origin == msg.sender || holds[operationId].issuer == msg.sender, "The hold can only be renewed by the issuer or the payer");
+
+        uint256 oldExpiration = holds[operationId].expiration;
+        if(timeToExpiration == 0){
+            holds[operationId].expires = false;
+            holds[operationId].expiration = 0;
+        }else{
+            holds[operationId].expires = true;
+            holds[operationId].expiration = timeToExpiration;
+        }
+
+        emit HoldRenewed(holds[operationId].issuer, operationId, oldExpiration, timeToExpiration);
+        return true;
+    }
+
+
+    function retrieveHoldData(string calldata operationId) external view returns (address from, address to, address notary, uint256 value, uint256 expiration, HoldStatusCode status){ //maybe also the issuer??
+
+        return (holds[operationId].origin, holds[operationId].target, holds[operationId].notary, holds[operationId].amount, holds[operationId].expiration, holds[operationId].status);
+    }
+
+
+    function balanceOnHold(address account) external view returns (uint256){
+        return heldBalance[account];
+    }
+
+
+    function netBalanceOf(address account) external view returns (uint256){
+        return balanceOf(account).sub(heldBalance[account]);
+    }
+
+
+    function totalSupplyOnHold() external view returns (uint256){
+        return _totalHeldBalance;
+    }
+
+
+    function authorizeHoldOperator(address operator) external returns (bool){
+
+        require (indexOperator[msg.sender][operator] == 0, "This operator is already authorized");
+
+        operatorsAuthorized[msg.sender].push(operator);
+        indexOperator[msg.sender][operator] = operatorsAuthorized[msg.sender].length;
+
+        emit AuthorizedHoldOperator(operator, msg.sender);
+        return true;
+    }
+
+
+    function revokeHoldOperator(address operator) external returns (bool){
+
+        require (indexOperator[msg.sender][operator] != 0, "This operator is already not authorized");
+
+        for (uint i = indexOperator[msg.sender][operator]-1; i<operatorsAuthorized[msg.sender].length-1; i++){
+            operatorsAuthorized[msg.sender][i] = operatorsAuthorized[msg.sender][i+1];
+            indexOperator[msg.sender][operatorsAuthorized[msg.sender][i]]--;
+        }
+        delete operatorsAuthorized[msg.sender][operatorsAuthorized[msg.sender].length-1];
+        operatorsAuthorized[msg.sender].length--;
+        indexOperator[msg.sender][operator] = 0;
+
+        emit RevokedHoldOperator(operator, msg.sender);
+        return true;
+    }
+
+    function isHoldOperatorFor(address operator, address from) external view returns (bool) {
+        // TODO: implementation is missing
+        return true;
+    }
+
+
```

Comment:
`transfer`, `transferFrom` and `balanceOf` need to be overwritten to take holds into account. Like for example here: https://gitlab.com/iobuilders/em-token/blob/master/contracts/HoldableToken.sol